sympy/parsing/latex/lark/latex_parser.py

```diff
@@ -0,0 +1,310 @@
+        s = s[:position] + replacement + s[position+1:]
+    return s
+
+
+def parse_latex(s):
+    r"""Converts the string ``s`` to a SymPy ``Expr``
+
+    Parameters
+    ==========
+
+    s : str
+        The LaTeX string to parse. In Python source containing LaTeX,
+        *raw strings* (denoted with ``r"``, like this one) are preferred,
+        as LaTeX makes liberal use of the ``\`` character, which would
+        trigger escaping in normal Python strings.
+
+    Examples
+    ========
+
+    >>> from sympy.parsing.latex import parse_latex
+    >>> expr = parse_latex(r"\frac {1 + \sqrt {\a}} {\b}")
+    >>> expr
+    (sqrt(a) + 1)/b
+    >>> expr.evalf(4, subs=dict(a=5, b=2))
+    1.618
+    """
+
+    # default behavior is to regenerate grammar from `LaTeX.lark` if
+    # `lark` is not importable it falls back to using the standalone
+    # precompiled lark latex grammar if it exists
+    _lark = import_module('lark')
+    if _lark is None:
+        # generated via lark standalone
+        # python -m lark.tools.standalone latex.lark > latex_grammar.py
+        standalone_lark = import_module('sympy.parsing.latex.latex_grammar')
+        if standalone_lark is None:
+            # TODO: throw proper error
+            raise ValueError('precompiled lark grammar "latex_grammar" not avialable')
+        parser = standalone_lark.Lark_StandAlone()
+        Transformer = standalone_lark.Transformer
+    else:
+        # TODO: should we use pkg_resource to get grammar file?  I
+        # think this would make sympy depend on setuptools which we
+        # would not like
+        with open(os.path.join(os.path.dirname(__file__), 'latex.lark')) as f:
+            latex_grammar = f.read()
+
+        parser = _lark.Lark(latex_grammar, parser='lalr',
+                            lexer='standard',
+                            propagate_positions=False,
+                            maybe_placeholders=False)
+        Transformer = _lark.Transformer
+
+    class TreeToSympy(Transformer):
+        INTEGER = int
+        FLOAT = float
+        LETTER = sympy.Symbol
+
+        def SUBSCRIPT_SYMBOL(self, token):
+            symbol, sub = token.value.replace('\\', '').split('_')
+            if sub.startswith('{'):
+                return sympy.Symbol('%s_{%s}' % (symbol, sub[1:-1]))
+            else:
+                return sympy.Symbol('%s_{%s}' % (symbol, sub))
+
+        def SYMBOL(self, token):
+            symbol = token.value.replace('\\', '')
+
+            if symbol == 'infty':
+                return sympy.oo
+            elif symbol == 'pi':
+                return sympy.pi
+            return sympy.Symbol(symbol)
+
+        def CMD_MATHIT(self, token):
+            return sympy.Symbol(token.value[8:-1])
+
+        def group(self, args):
+            print(args)
+            if len(args) == 5: # e.g. \\left ( expr \right )
+                return args[2]
+            else: # e.g. ( expr )
+                return args[1]
```

Comment:
You can avoid indexing if you use `_` prefix in terminal
https://lark-parser.readthedocs.io/en/latest/tree_construction.html#terminals